### PR TITLE
prepare new release

### PR DESCRIPTION
**Pull request by Byron**
This time, the release has a changelog and I will try to use `cargo smart-release` to do it.

### CODE REVIEW

**Comment by Byron** [comment]
Hi @joshtriplett, could you review this change? I forgot that I can't self-approve here, and thought it was over at `libz-sys`.
It's notable that I already created a release from this branch, and a tag was pushed as well.

**Comment by badboy** [comment]
There's now a [1.0.31 release on crates.io](https://crates.io/crates/flate2/1.0.31), but no tag in this repository nor a changelog until this is merged.

**Comment by Byron** [comment]
That's correct - this PR is waiting for approval so it can be merged.

**Review by joshtriplett — APPROVED** [review]

**Pull request merged**
